Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+                                                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                                        @{N='ResourceName';E={$_.ResourceName}},
+                                                                                        @{N='ListOfAPIRolledBack';E={$listOfApiRolledBack}}
+            }
+
+            if($listOfApiSkipped.count -ne '0')
+            {
+                $listOfApiSkipped = $listOfApiSkipped -join ","
+                $apiManagementServicesSkipped += $apiManagementservice | Select-Object @{N='ResourceID';E={$_.ResourceId}},
+                                                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                                        @{N='ResourceName';E={$_.ResourceName}},
+                                                                                        @{N='ListOfAPISkipped';E={$listOfApiSkipped}}
+            }        
+        }
+        catch
+        {
+            $listOfApiSkipped = $listOfApiSkipped -join ","
+            $apiManagementServicesSkipped += $apiManagementservice | Select-Object @{N='ResourceID';E={$_.ResourceId}},
+                                                                                    @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                                    @{N='ResourceName';E={$_ResourceName}},
+                                                                                    @{N='ListOfAPISkipped';E={$listOfApiSkipped}}
+            Write-Host "Error enabling HTTPS url scheme on the API Management service. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping this API Management Service. HTTPS url scheme will not be enabled for any of the API(s) of this service." -ForegroundColor $([Constants]::MessageType.Error)
+            return
+        }  
+    }  
+
+    if ($($apiManagementServicesRolledBack | Measure-Object).Count -gt 0 -or $($apiManagementServicesSkipped | Measure-Object).Count -gt 0)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "Rollback Summary:`n" -ForegroundColor $([Constants]::MessageType.Info)
+        
+        if ($($apiManagementServicesRolledBack | Measure-Object).Count -gt 0)
+        {
+            Write-Host "HTTPS url scheme successfully disabled for the API(s) of following API Management Service(s):" -ForegroundColor $([Constants]::MessageType.Update)
+            $apiManagementServicesRolledBack | Format-Table -Property ResourceGroupName , ResourceName , ListOfAPIRolledBack
+
+            # Write this to a file.
+            $apiManagementServiceRolledBackFile = "$($backupFolderPath)\RolledBackApiManagementServices.csv"
```

Comment:
Ideally it is this file name convention that needs to be followed everywhere. Please use this as a reference for your changes.